File: backend/src/models/Vehicle.js
const mongoose = require('mongoose');

const vehicleSchema = new mongoose.Schema({
  brand: {
    type: String,
    required: true,
    trim: true
  },
  model: {
    type: String,
    required: true,
    trim: true
  },
  year: {
    type: String,
    required: true,
    trim: true
  },
  engineType: {
    type: String,
    required: true,
    enum: ['Bencina', 'Petróleo'],
    default: 'Bencina'
  },
  engineSize: {
    type: String,
    required: true,
    trim: true
  },
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  }
}, {
  timestamps: true
});

module.exports = mongoose.model('Vehicle', vehicleSchema);

File: frontend/lib/models/vehicle.dart
class Vehicle {
  String? id;
  String brand;
  String model;
  String year;
  String engineType;
  String engineSize;
  String userId;

  Vehicle({
    this.id,
    required this.brand,
    required this.model,
    required this.year,
    required this.engineType,
    required this.engineSize,
    required this.userId,
  });

  factory Vehicle.fromJson(Map<String, dynamic> json) {
    return Vehicle(
      id: json['_id'] ?? json['id'],
      brand: json['brand'],
      model: json['model'],
      year: json['year'],
      engineType: json['engineType'],
      engineSize: json['engineSize'],
      userId: json['userId'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'brand': brand,
      'model': model,
      'year': year,
      'engineType': engineType,
      'engineSize': engineSize,
      'userId': userId,
    };
  }
}
